### PR TITLE
Cleanup the code of collecting test cases from XML file

### PR DESCRIPTION
**Pull request by simonxiaoss**
1. Cleanup the code and make it more readable
2. Support * to filter test cases to run

PS C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\LISAv2> ./Run-LisaV2.ps1 -TestPlatform "Azure" -TestLocation "westus2" -RGIdentifier "$env:UserName" -ARMImageName "Canonical
 UbuntuServer 16.04-LTS latest" -TestNames "*" -XMLSecretFile "C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\AzureSecrets.xml" -ForceDeleteResources -TestCategory * -TestArea
 * -TestTag core -TestPriority 0
Test ID generated for this test run: ESVC53721
Current working directory 'C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\LISAv2' path length is too long.
Copying current workspace to C:\LISAv2\ESVC53721
Working directory has been changed to C:\LISAv2\ESVC53721
11/28/2018 00:39:50 : [INFO ] Created LogDir: C:\LISAv2\ESVC53721\TestResults\2018-27-11-16-39-50-7431
11/28/2018 00:39:50 : [INFO ] Test parameters have been validated successfully. Continue running the test.
11/28/2018 00:39:51 : [INFO ] Updated GlobalConfigurations.xml file.
11/28/2018 00:39:51 : [INFO ] Updated Test Case xml files.
11/28/2018 00:39:51 : [INFO ] Using provided secrets file: C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\AzureSecrets.xml
11/28/2018 00:39:51 : [INFO ] C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\AzureSecrets.xml found.
11/28/2018 00:39:51 : [INFO ] ------------------------------------------------------------------
11/28/2018 00:39:51 : [INFO ] Authenticating Azure PS session..
11/28/2018 00:39:52 : [INFO ] Current Subscription : xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.
11/28/2018 00:39:52 : [INFO ] ------------------------------------------------------------------
11/28/2018 00:39:52 : [INFO ] Validating XML Files from C:\LISAv2\ESVC53721 folder recursively...
11/28/2018 00:39:53 : [INFO ] Collected: BVT-CORE-TIMESYNC-NTP
11/28/2018 00:39:53 : [INFO ] Collected: BVT-CORE-VERIFY-LIS-MODULES
11/28/2018 00:39:53 : [INFO ] Collected: BVT-CORE-VERIFY-LIS-MODULES-VERSION
11/28/2018 00:39:53 : [INFO ] Collected: BVT-CORE-RELOAD-MODULES-SMP
11/28/2018 00:39:53 : [INFO ] Collected: CORE-VERIFY-LIS-MODULES-VERSION
11/28/2018 00:39:53 : [INFO ] 5 Test Cases have been collected
11/28/2018 00:39:53 : [INFO ] Test cases are scanned and imported to C:\LISAv2\ESVC53721\TestConfiguration.xml
11/28/2018 00:39:53 : [INFO ] The auto created C:\LISAv2\ESVC53721\TestConfiguration.xml has been validated successfully.
11/28/2018 00:39:53 : [INFO ] .\AutomationManager.ps1 -xmlConfigFile 'C:\LISAv2\ESVC53721\TestConfiguration.xml' -cycleName 'TC-ESVC53721' -RGIdentifier 'simonxiaoss' -TestReport
XmlPath C:\LISAv2\ESVC53721\Report\LISAv2_TestReport_ESVC53721.xml -ForceDeleteResources -XMLSecretFile 'C:\Users\simonxiaoss\Documents\GitHub\simonxiaoss\AzureSecrets.xml'

### CODE REVIEW

File: Libraries/LISAV2-Framework.psm1 [hunk 1]
```diff
@@ -295,158 +299,96 @@ Function Match-TestPriority($currentTest)
...
-                            Write-LogInfo "Collected $($test.TestName)"
-                            $AllLisaTests += $test
-                        }
-                    }
-                }
+            if (($test.Area -ne $TestArea) -and ($TestArea -ne "*")) {
+                continue
+            }
+
+            if (($test.TestNames -ne $TestNames) -and ($TestNames -ne "*")) {
```
Comment (LiliDeng) [inline Libraries/LISAV2-Framework.psm1]: $test.TestNames => $test.testName

Comment (lizzha) [reply Libraries/LISAV2-Framework.psm1]: the difference is the extra "s" in the end

**Comment by LiliDeng** [comment]
Current it supports below format that is run more than 1 case for each run.
 _-TestNames "BVT-SWAP-CHECK,BVT-VERIFY-DEPLOYMENT-PROVISION"_ 
In this PR, it doesn't support any more.

File: Libraries/LISAV2-Framework.psm1 (hunk 1 above)
Comment (simonxiaoss) [reply Libraries/LISAV2-Framework.psm1]: Good catch. Will fix.

File: Libraries/LISAV2-Framework.psm1 [hunk 2]
```diff
@@ -276,15 +276,19 @@ Function Update-XMLStringsFromSecretsFile($XmlSecretsFilePath)
...
     }
 
+    if( $TestPriority -eq "*") {
+        return $True
+    }
+
     $priorityInXml = $currentTest.Priority
     if (-not $priorityInXml) {
-        Write-LogInfo "Warning: Priority of $($currentTest.TestName) is not defined, set its priority 1 by default."
+        Write-LogWarn "Warning: Priority of $($currentTest.TestName) is not defined, set it to 1 (default priority)."
```
[the remark this answers is not in the document]
Comment (simonxiaoss) [reply Libraries/LISAV2-Framework.psm1]: Will fix.

**Pull request merged**